token/transfer-hook/cli/src/main.rs

```diff
@@ -0,0 +1,405 @@
+    solana_sdk::{
+        commitment_config::CommitmentConfig,
+        instruction::AccountMeta,
+        pubkey::Pubkey,
+        signature::{Signature, Signer},
+        system_instruction,
+        transaction::Transaction,
+    },
+    spl_tlv_account_resolution::state::ExtraAccountMetaList,
+    spl_transfer_hook_interface::{
+        get_extra_account_metas_address, instruction::initialize_extra_account_meta_list,
+    },
+    std::{fmt, process::exit, rc::Rc, str::FromStr},
+    strum_macros::{EnumString, IntoStaticStr},
+};
+
+#[derive(Debug, Clone, Copy, PartialEq, EnumString, IntoStaticStr)]
+#[strum(serialize_all = "kebab-case")]
+pub enum AccountMetaRole {
+    Readonly,
+    Writable,
+    ReadonlySigner,
+    WritableSigner,
+}
+impl fmt::Display for AccountMetaRole {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{:?}", self)
+    }
+}
+fn parse_transfer_hook_account(arg: &str) -> Result<AccountMeta, String> {
+    match arg.split(':').collect::<Vec<_>>().as_slice() {
+        [address, role] => {
+            let address = Pubkey::from_str(address).map_err(|e| format!("{e}"))?;
+            let meta = match AccountMetaRole::from_str(role).map_err(|e| format!("{e}"))? {
+                AccountMetaRole::Readonly => AccountMeta::new_readonly(address, false),
+                AccountMetaRole::Writable => AccountMeta::new(address, false),
+                AccountMetaRole::ReadonlySigner => AccountMeta::new_readonly(address, true),
+                AccountMetaRole::WritableSigner => AccountMeta::new(address, true),
+            };
+            Ok(meta)
+        }
+        _ => Err("Transfer hook account must be present as <ADDRESS>:<ROLE>".to_string()),
+    }
+}
+
+fn clap_is_valid_pubkey(arg: &str) -> Result<(), String> {
+    is_valid_pubkey(arg)
+}
+
+struct Config {
+    commitment_config: CommitmentConfig,
+    default_signer: Box<dyn Signer>,
+    json_rpc_url: String,
+    verbose: bool,
+}
+
+async fn process_create_extra_account_metas(
+    rpc_client: &RpcClient,
+    program_id: &Pubkey,
+    token: &Pubkey,
+    transfer_hook_accounts: Vec<AccountMeta>,
+    mint_authority: &dyn Signer,
+    payer: &dyn Signer,
+) -> Result<Signature, Box<dyn std::error::Error>> {
+    let extra_account_metas_address = get_extra_account_metas_address(token, program_id);
+    let extra_account_metas = transfer_hook_accounts
+        .into_iter()
+        .map(|v| v.into())
+        .collect::<Vec<_>>();
+
+    let length = extra_account_metas.len();
+    let account_size = ExtraAccountMetaList::size_of(length)?;
+    let required_lamports = rpc_client
+        .get_minimum_balance_for_rent_exemption(account_size)
+        .await
+        .map_err(|err| format!("error: unable to fetch rent-exemption: {err}"))?;
+    let current_lamports = rpc_client
+        .get_account(&extra_account_metas_address)
+        .await
+        .map(|a| a.lamports)
+        .unwrap_or(0);
```

Comment:
Should we maybe have a check in case the account exists?

Reply:
I know the `init_extension` does this, but just curious if we want it on the CLI, too.

Reply:
Ah good call, yeah I'll add the check